### PR TITLE
Update AuthenticationKey to extend from Serializable

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Update AuthKey to extend from Serializable.

### Test Plan

pnpm test

### CODE REVIEW

**Comment by 0xjinn** [comment]
Current dependencies on/for this PR:
* main
  * **PR #148** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/148" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-ts-sdk/148?utm_source=stack-comment).

**Review by 0xmaayan — APPROVED** [review]

File: tests/unit/authenticationKey.test.ts [hunk 1]
```diff
@@ -91,4 +91,13 @@ describe("AuthenticationKey", () => {
...
   });
+
+  it("should serialize and deserialize correctly", () => {
+    const authKey = new AuthenticationKey({ data: ed25519.authKey });
+    const serializer = new Serializer();
+    authKey.serialize(serializer);
+    const deserializer = new Deserializer(serializer.toUint8Array());
+    const authKeyDeserialized = AuthenticationKey.deserialize(deserializer);
+    expect(authKeyDeserialized.data.toString()).toEqual(ed25519.authKey);
+  });
```
Comment (gregnazario) [inline tests/unit/authenticationKey.test.ts:102]: FYI for a test like this, you should ensure that the data matches what you expect in bytes if it's supposed to be consumed by anything else.

You can serialize and deserialize something wrong, as long as it's symmetric :)

**Review by xbtmatt — APPROVED** [review]
I think just add a more meaningful test (should be easy since it's a simple auth key) and it lgtm

File: tests/unit/authenticationKey.test.ts (hunk 1 above)
Comment (0xjinn) [reply tests/unit/authenticationKey.test.ts:102]: Ah I kept forgetting about this. Will update soon

**Comment by 0xjinn** [comment]
Update: Split the test int Serialize and Deserialize separately

**Pull request merged**